cray_product_catalog/catalog_delete.py

```diff
@@ -165,17 +183,42 @@ def main():
     # Parameters to identify ConfigMap and product/version to remove
     PRODUCT = os.environ.get("PRODUCT").strip()  # required
     PRODUCT_VERSION = os.environ.get("PRODUCT_VERSION").strip()  # required
-    CONFIG_MAP = os.environ.get("CONFIG_MAP", "cray-product-catalog").strip()
     CONFIG_MAP_NS = os.environ.get("CONFIG_MAP_NAMESPACE", "services").strip()
+    PRODUCT_CONFIG_MAP = format_product_cm_name(CONFIG_MAP, PRODUCT)
     KEY = os.environ.get("KEY", "").strip() or None
 
-    args = (CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, KEY)
-    LOGGER.info(
-        "Removing from ConfigMap=%s in namespace=%s for %s/%s (key=%s)",
-        *args
-    )
     load_k8s()
-    modify_config_map(*args)
+
+    if KEY:
+        if KEY in CONFIG_MAP_FIELDS:
+            args = (CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, KEY, MAX_RETRIES)
+        elif KEY in PRODUCT_CM_FIELDS:
+            args = (PRODUCT_CONFIG_MAP, CONFIG_MAP_NS, PRODUCT, PRODUCT_VERSION, KEY, MAX_RETRIES_FOR_PROD_CM)
```

Comment:
`format_product_cm_name` can return empty string ''.. Shall we put a check for non-empty and only then go forward. Also add a error log that Product CM generated is empty, cannot go for deletion?